src/coreclr/clrfeatures.cmake

```diff
@@ -3,7 +3,11 @@ if(CLR_CMAKE_TARGET_TIZEN_LINUX)
 endif()
 
 if(NOT DEFINED FEATURE_EVENT_TRACE)
-  set(FEATURE_EVENT_TRACE 1)
+  if (CLR_CMAKE_BUILD_SUBSET_JIT)
```

Comment:
Subsets should control which parts of the tree get built; not how they are built.

Should this be conditionalized based on the platform instead?

Reply:
The reason I disabled this is that code enabled by FEATURE_EVENT_TRACE relies on code that is only build when the JIT is built.   E.g. it includes the "clrprovider.h" header that will only get generated when the JIT is built.

Should I be looking for some way to make this work without the JIT?

Reply:
Yes, the ILTools subset should be independent on the JIT.

(Also, the JIT should not depend on `FEATURE_EVENT_TRACE` that's another accidental dependency that should not be there.)

Reply:
Ok, I've looked a bit closer into the dependency chains.  The original problem was that the IL tools depend on utilcode, which in turn depends on eventing_headers, which uses the clrprovider.h header that is only built as part of the JIT.

However, looking more closely, the utilcode directory actually builds four different libraries: utilcode, utilcode_dac, utilcode_crossgen, and utilcodestaticnohost.  And only the first of those (utilcode) has a **real** dependency on the event stuff, the other three only have spurious dependencies as the header file is included but never actually used.

Plus, the IL tools only require utilcodestaticnohost, not the main utilcode.   So it seems this can be properly fixed by:
- removing the spurious dependencies for utilcode_dac, utilcode_crossgen and utilcodestaticnohost; and
- only building the main utilcode library when CLR_CMAKE_BUILD_SUBSET_JIT is true

Does this make sense?   I'll try out whether this works for me.

Reply:
That didn't quite work, sorry.  Turns out we already need the eventing headers to build the paltests when FEATURE_EVENT_TRACE is on.

However, we can actually just always build the headers, even when not building the JIT, by including vm/eventing (without its eventpipe subdirectory) in the build.  I've pushed an updated patch that uses that approach.